fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonExpressionConverter.java

```diff
@@ -0,0 +1,214 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.starrocks.connector.paimon;
+
+import com.starrocks.analysis.BinaryPredicate;
+import com.starrocks.analysis.BoolLiteral;
+import com.starrocks.analysis.CastExpr;
+import com.starrocks.analysis.CompoundPredicate;
+import com.starrocks.analysis.DateLiteral;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FloatLiteral;
+import com.starrocks.analysis.InPredicate;
+import com.starrocks.analysis.IntLiteral;
+import com.starrocks.analysis.IsNullPredicate;
+import com.starrocks.analysis.LiteralExpr;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.StringLiteral;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import com.starrocks.sql.ast.AstVisitor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.paimon.data.BinaryString;
+import org.apache.paimon.predicate.Predicate;
+import org.apache.paimon.predicate.PredicateBuilder;
+import org.apache.paimon.types.DataType;
+import org.apache.paimon.types.RowType;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.paimon.data.Timestamp.fromSQLTimestamp;
+
+public class PaimonExpressionConverter extends AstVisitor<Predicate, Void> {
+    private static final Logger LOG = LogManager.getLogger(PaimonExpressionConverter.class);
+
+    private final RowType rowType;
+    private final PredicateBuilder builder;
+    private final List<String> fieldNames;
+
+    public PaimonExpressionConverter(RowType rowType, List<String> fieldNames) {
+        this.rowType = rowType;
+        this.builder = new PredicateBuilder(rowType);
+        this.fieldNames = fieldNames;
+    }
+
+    public Predicate convert(Expr expr) {
+        if (expr == null) {
+            return null;
+        }
+        return visit(expr);
+    }
+
+    @Override
+    public Predicate visitCompoundPredicate(CompoundPredicate node, Void context) {
```

Comment:
It's better to pass `ScalarOperator` instead of `Expr` because you will need to get the predicate for statistics later.